plugins/git/current_git.mli

```diff
@@ -39,10 +39,19 @@ module Commit : sig
   val unmarshal : string -> t
 end
 
-val clone : schedule:Current_cache.Schedule.t -> ?gref:string -> string -> Commit.t Current.t
-(** [clone ~schedule ~gref uri] evaluates to the head commit of [uri]'s [gref] branch (default: "master"). *)
-
-val fetch : Commit_id.t Current.t -> Commit.t Current.t
+val clone :
+  schedule:Current_cache.Schedule.t ->
+  ?token:(string, [ `Msg of string ]) result Lwt.t  ->
+  ?gref:string ->
+  string ->
+  Commit.t Current.t
+(** [clone ~schedule ?token ~gref uri] evaluates to the head commit of [uri]'s [gref] branch (default: "master").
+    [token] is an authentification token you can provide if the forge supports API tokens and private repositories. *)
```

Comment:
This seems to be GitHub specific it would be worth noting that in the comments. We could support GitLab style cloning which uses a url scheme of `git clone https://<username>:<personal_token>@gitlab.com/gitlab-org/gitlab.git` with some further work.

https://docs.gitlab.com/ee/user/profile/personal_access_tokens.html#clone-repository-using-personal-access-token